Documentation/contributing/testing/e2e.rst

```diff
@@ -40,6 +40,236 @@ determine which VMs are necessary to run particular tests. All test names
+
+        $ ssh -p 2222 -o "StrictHostKeyChecking=no" root@localhost
+        # cd /host/test
+        # kernel_tag="bpf-next-20230526.105339@sha256:4133d4e09b1e86ac175df8d899873180281bb4220dc43e2566c47b0241637411"
+        # k8s_version="1.27"
+        #
+        # export K8S_NODES=2
+        # export NETNEXT=0
+        # export K8S_VERSION="${k8s_version}"
+        # export CNI_INTEGRATION=kind
+        # export INTEGRATION_TESTS=true
+        #
+        # if [[ "${kernel_tag}" == bpf-next-* ]]; then
+        #    export KERNEL=net-next
+        #    export NETNEXT=1
+        #    export KUBEPROXY=0
+        #    export K8S_NODES=3
+        #    export NO_CILIUM_ON_NODES=kind-worker2
+        # elif [[ "${kernel_tag}" == 4.19-* ]]; then
+        #    export KERNEL=419
+        # elif [[ "${kernel_tag}" == 5.4-* ]]; then
+        #    export KERNEL=54
+        # fi
+        #
+        # # GitHub actions do not support IPv6 connectivity to outside
+        # # world. If the infrastructure environment supports it, then
+        # # this line can be removed
+        # export CILIUM_NO_IPV6_OUTSIDE=true
+        #
+        # commit_sha="7b368923823e63c9824ea2b5ee4dc026bc4d5cd8"
+        # cliFocus="K8sDatapathConfig Check|K8sDatapathConfig IPv4Only|K8sDatapathConfig High-scale|K8sDatapathConfig Iptables|K8sDatapathConfig IPv4Only|K8sDatapathConfig IPv6|K8sDatapathConfig Transparent"
+        # quay_org="cilium"
+        #
+        # ./test.test \
+          --ginkgo.focus="${cliFocus}" \
+          --ginkgo.skip="" \
+          --ginkgo.seed=1679952881 \
+          --ginkgo.v -- \
+          -cilium.provision=false \
+          -cilium.image=quay.io/${quay_org}/cilium-ci \
+          -cilium.tag=${commit_sha}  \
+          -cilium.operator-image=quay.io/${quay_org}/operator \
+          -cilium.operator-tag=${commit_sha} \
+          -cilium.hubble-relay-image=quay.io/${quay_org}/hubble-relay-ci \
+          -cilium.hubble-relay-tag=${commit_sha} \
+          -cilium.kubeconfig=/root/.kube/config \
+          -cilium.provision-k8s=false \
+          -cilium.operator-suffix=-ci \
+          -cilium.holdEnvironment=true
+        Using CNI_INTEGRATION="kind"
+        Running Suite: Suite-k8s-1.27
+        =============================
+        Random Seed: 1679952881
+        Will run 7 of 132 specs
+
+7. Wait until the test execution completes
+
+   .. code-block:: shell-session
+
+        Ran 4 of 132 Specs in 721.007 seconds
+        SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 128 Skipped
+
+8. Clean up
+
+   Once tests are performed, qemu can be terminated by checking the PID and
+   terminate the process.
+
+   .. code-block:: shell-session
+
+        $ pidof qemu-system-x86_64
+        1593413
+        $ kill 1593413
+
+   The VM state is kept in ``/tmp/_images/datapath-conformance.qcow2`` and the
+   dependencies are installed. Thus steps 1, 2, 3 and 4 can be skipped
+   next time and the VM state can be re-used from step 5 onwards.
```

Comment:
Sorry, we talked about it offline but I was not clear enough. In my opinion the options here are:

- Use numbers everywhere, in the list and below to reference specific list items
- Nowhere (automatic numbering like you do now), and reference items with hyperlinks

The first approach is fragile, but very easy for the readers. The second is robust, but it's maybe slightly less clear to understand immediately what step we're referring to.

With the current mix, if we insert a new step in the middle of the list, the references at the bottom will “break”, and it will be harder to track because we won't see numbers change in the git history. Could you please stick to numbers everywhere, or no number at all?

Reply:
Fixed!